src/main/java/com/edmunds/rest/databricks/DTO/scim/user/UserDTO.java

```diff
@@ -0,0 +1,228 @@
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Databricks SCIM user.
+ * @see <a href="https://tools.ietf.org/html/rfc7643#section-4.1">https://tools.ietf.org/html/rfc7643#section-4.1</a>
+ */
+public class UserDTO {
+
+  @JsonProperty("schemas")
+  private final String[] schemas = new String[]{"urn:ietf:params:scim:schemas:core:2.0:User"};
+  @JsonProperty("id")
+  private long id;
+  @JsonProperty("userName")
+  private String userName;
+  @JsonProperty("name")
+  private NameDTO name;
+  @JsonProperty("emails")
+  private EmailDTO[] emails;//can't be actually customized. setting username will create an entry (work,username,primary)
+  @JsonProperty("active")
+  private boolean active = true;
+  @JsonSerialize(using = UserGroupSerializer.class)
+  @JsonDeserialize(using = UserGroupDeSerializer.class)
+  private GroupDTO[] groups = new GroupDTO[0];
+  @JsonProperty("entitlements")
+  private EntitlementsDTO[] entitlements = new EntitlementsDTO[0];
+  @JsonProperty("displayName")
+  private String displayName;
+
+  public UserDTO() {
+  }
+
+  public UserDTO(UserDTO from) {
+    this.id = from.id;
+    this.userName = from.userName;
+    if (from.name != null) {
+      this.name = new NameDTO(from.name);
+    }
+    if (from.emails != null) {
+      emails = new EmailDTO[from.emails.length];
+      for (int i = 0; i < emails.length; i++) {
+        emails[i] = new EmailDTO(from.emails[i]);
+      }
+    }
+
+    this.active = from.active;
+    if (from.groups != null) {
+      groups = new GroupDTO[from.groups.length];
+      for (int i = 0; i < groups.length; i++) {
+        groups[i] = new GroupDTO(from.groups[i]);
+      }
+    }
+    if (from.entitlements != null) {
+      entitlements = new EntitlementsDTO[from.entitlements.length];
+      for (int i = 0; i < entitlements.length; i++) {
+        entitlements[i] = new EntitlementsDTO(from.entitlements[i]);
+      }
+    }
+    this.displayName = from.displayName;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    UserDTO userDTO = (UserDTO) o;
+    return active == userDTO.active &&
+        Objects.equals(userName, userDTO.userName) &&
+        Objects.equals(name, userDTO.name) &&
+        Arrays.equals(emails, userDTO.emails) &&
+        Arrays.equals(groups, userDTO.groups) &&
+        Arrays.equals(entitlements, userDTO.entitlements);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = Objects.hash(userName, name, active);
+    result = 31 * result + Arrays.hashCode(emails);
+    result = 31 * result + Arrays.hashCode(groups);
+    result = 31 * result + Arrays.hashCode(entitlements);
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "UserDTO{" +
```

Comment:
I'd prefer to see commons-lang StringUtils, but, I'm not sure what @samshuster thinks, and if it is consistent with the rest of the project.

Reply:
looks like we don't have that dependency yet. Would you want to add it? Your call @javamonkey79

Reply:
If it's not already there, then I wouldn't want to add it at this point; also, it wouldn't be consistent with the other toString methods. So... no :)